custom_metrics/markup.js

```diff
@@ -0,0 +1,580 @@
+                        break;
+                    case "":
+                        result.loading.blank++;
+                        break;
+                    default:
+                        result.loading.invalid++;
+                        break;
+                }
+            }
+            else {
+                result.loading.missing++;
+            }
+        });
+
+        return result;
+      }
+      catch(e) {
+        return logError("iframes", e);
+      }
+    })(),
+
+    // amp related data
+    // Used by: SEO  
+    'amp': (() => {  
+      try {
+        let result = {};
+
+        // valid amp page
+        // https://amp.dev/documentation/guides-and-tutorials/learn/spec/amphtml/
+        // doctype is html
+        // html amp or ⚡ attribute
+        // canonical link to real page
+        // <meta charset="utf-8">
+        // <meta name="viewport" content="width=device-width">
+        // < script async src="https://cdn.ampproject.org/v0.js"></ script>
+        // boilerplate css
+
+        result.html_amp_attribute_present = !!document.querySelector('html')?.hasAttribute('amp');
+
+        result.html_amp_emoji_attribute_present = !!document.querySelector('html')?.hasAttribute('⚡');
+
+        result.amp_page = result.html_amp_attribute_present || result.html_amp_emoji_attribute_present;
+
+        result.rel_amphtml = document.querySelector("link[rel='amphtml']")?.getAttribute('href') ?? null;
+
+        var metadata = document.querySelector('meta[name=generator][content^="AMP Plugin"]');
+        if (metadata) {
+          result.amp_plugin =  metadata.getAttribute('content') ?? null;
+        }
+
+        return result;
+      }
+      catch(e) {
+        return logError("amp", e);
+      }
+    })(),
+
+    // data-nosnippet use 
+    // Used by: SEO
+    'data_nosnippet': (() => { 
+      try {    
+        // https://support.google.com/webmasters/answer/79812?hl=en
+        // https://developers.google.com/search/reference/robots_meta_tag
+        var validNodes = document.querySelectorAll('span[data-nosnippet], div[data-nosnippet], section[data-nosnippet]');
+        var allNodes = document.querySelectorAll('[data-nosnippet]');
+        return { valid: validNodes.length, wrong_tag_type: allNodes.length - validNodes.length};
+      }
+      catch(e) {
+        return logError("data_nosnippet", e);
+      }
+    })(),
+
+    // markup info 
+    // Used by: Markup
+    'obsolete_elements': (() => {  
+      try {   
+        // https://html.spec.whatwg.org/multipage/obsolete.html#non-conforming-features
+        // Array.from(document.querySelectorAll('dfn[data-dfn-type] code')).map(e => e.innerText).join(',')
+        let result = {};
+        let obsoleteNodes = [...document.querySelectorAll('applet,acronym,bgsound,dir,frame,frameset,noframes,isindex,keygen,listing,menuitem,nextid,noembed,plaintext,rb,rtc,strike,xmp,basefont,big,blink,center,font,marquee,multicol,nobr,spacer,tt')];
```

Comment:
Please break this line up into multiple lines
```suggestion
        let obsoleteNodes = [...document.querySelectorAll(
          'applet,acronym,bgsound,dir,frame,frameset,noframes,isindex,keygen,listing,menuitem,nextid,noembed,' +
          'plaintext,rb,rtc,strike,xmp,basefont,big,blink,center,font,marquee,multicol,nobr,spacer,tt'
        )];
```

Reply:
Bonus points if you alphabetize the list, so its easier to see what is and is not included